From 96f8c48ceccabb3fb9d0ddcc3a7b180c2ae4a4b3 Mon Sep 17 00:00:00 2001
From: Shashank Sharma <shashank.sharma@intel.com>
Date: Mon, 22 Apr 2019 15:22:50 +0530
Subject: [PATCH 25/27] compositor-drm: Allow HDR surfaces in render-only mode
 only

As the display sw stack doesn't have support to handle tone-mapping
and color corrections for HDR surfaces, do not allow HDR view composition
in plane-only/mixed mode, force it in render-only composition mode.

Signed-off-by: Shashank Sharma <shashank.sharma@intel.com>
Signed-off-by: alex.chae <alex.chae@intel.com>
---
 libweston/backend-drm/state-propose.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/backend-drm/state-propose.c b/libweston/backend-drm/state-propose.c
index d2729377..9d7cf670 100644
--- a/libweston/backend-drm/state-propose.c
+++ b/libweston/backend-drm/state-propose.c
@@ -1245,7 +1245,7 @@ drm_assign_planes(struct weston_output *output_base, void *repaint_data)
 		}
 	}
 
-	if (!b->sprites_are_broken && !output->virtual) {
+	if (!b->sprites_are_broken && !output->virtual && !hdr_surface) {
 		drm_debug(b, "\t[repaint] trying planes-only build state\n");
 		state = drm_output_propose_state(output_base, pending_state, mode);
 		if (!state) {
-- 
2.21.3

